gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/InMemoryTopology.java

```diff
@@ -0,0 +1,140 @@
+package gobblin.config.common.impl;
+
+import gobblin.config.store.api.ConfigKeyPath;
+
+import java.util.Collection;
+import java.util.List;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ListMultimap;
+
+/**
+ * InMemoryTopology will return stale data if the internal config store is Not {@ConfigStoreWithStableVersioning}
+ * 
+ * @author mitu
+ *
+ */
+public class InMemoryTopology implements ConfigStoreTopologyIntf {
+
+  private final ConfigStoreBackedTopology fallback;
+
+  private final ListMultimap<ConfigKeyPath, ConfigKeyPath> childrenMap = ArrayListMultimap.create();
+  private final ListMultimap<ConfigKeyPath, ConfigKeyPath> ownImportMap = ArrayListMultimap.create();
+  private final HashMultimap<ConfigKeyPath, ConfigKeyPath> ownImportedByMap = HashMultimap.create();
+  private final ListMultimap<ConfigKeyPath, ConfigKeyPath> recursiveImportMap = ArrayListMultimap.create();
+  private final HashMultimap<ConfigKeyPath, ConfigKeyPath> recursiveImportedByMap = HashMultimap.create();
+
+  public InMemoryTopology(ConfigStoreBackedTopology fallback) {
```

Comment:
The fallback should be any ConfigStoreTopology implementation not only ConfigStoreBackedTopology